src/app/kvm/components/SettingsBackLink/SettingsBackLink.tsx

```diff
@@ -1,15 +1,14 @@
 import { Icon } from "@canonical/react-components";
-import type { Location } from "history";
 import { Link, useLocation } from "react-router-dom";
 
 type LocationState = {
   from?: string;
 };
 
 const SettingsBackLink = (): JSX.Element | null => {
-  const location = useLocation<Location>();
+  const location = useLocation();
   const state = location.state as LocationState;
-  if (state === undefined || !state.from) {
+  if (!state?.from || state === undefined) {
```

Comment:
Nitpicking: this can be made more concise and easier to read. Optional chaining short circuits and evaluates to undefined if state is undefined anyway, no need to check it separately.

```suggestion
  if (!state?.from) {
```